src/analysis/retail/priest/holy/modules/features/HolyPriestHealingEfficiencyTracker.tsx

```diff
@@ -86,28 +132,31 @@ class HolyPriestHealingEfficiencyTracker extends HealingEfficiencyTracker {
   }
 
   addEcho(spellInfo: any, healingSpellIds: number[]) {
-    if (this.echoOfLight.masteryHealingBySpell[spellInfo.spell.id]) {
-      spellInfo.healingDone += this.echoOfLight.masteryHealingBySpell[
-        spellInfo.spell.id
-      ].effectiveHealing;
-      spellInfo.overhealingDone += this.echoOfLight.masteryHealingBySpell[
-        spellInfo.spell.id
-      ].overHealing;
-    }
+    try {
+      if (this.echoOfLight.masteryHealingBySpell[spellInfo.spell.id]) {
+        spellInfo.healingDone += this.echoOfLight.masteryHealingBySpell[
+          spellInfo.spell.id
+        ].effectiveHealing;
+        spellInfo.overhealingDone += this.echoOfLight.masteryHealingBySpell[
+          spellInfo.spell.id
+        ].overHealing;
+      }
 
-    if (healingSpellIds) {
-      healingSpellIds.forEach((healingSpellId) => {
-        if (this.echoOfLight.masteryHealingBySpell[healingSpellId]) {
-          spellInfo.healingDone += this.echoOfLight.masteryHealingBySpell[
-            healingSpellId
-          ].effectiveHealing;
-          spellInfo.overhealingDone += this.echoOfLight.masteryHealingBySpell[
-            healingSpellId
-          ].overHealing;
-        }
-      });
+      if (healingSpellIds) {
+        healingSpellIds.forEach((healingSpellId) => {
+          if (this.echoOfLight.masteryHealingBySpell[healingSpellId]) {
+            spellInfo.healingDone += this.echoOfLight.masteryHealingBySpell[
+              healingSpellId
+            ].effectiveHealing;
+            spellInfo.overhealingDone += this.echoOfLight.masteryHealingBySpell[
+              healingSpellId
+            ].overHealing;
+          }
+        });
+      }
+    } catch {
+      return spellInfo; //Avoids crashes
```

Comment:
What crashes were occurring?

Reply:
I am not sure the exact reason but it passed empty spellinfo with some different spells like dispel magic and divine star where they weren't used. No healing is lost so I figured I might as well fix it like this.